Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs

```diff
@@ -1,16 +1,323 @@
+                .SetOutput("reduced", "magnitude_squared"))
+              .Operators.Add(new Operator("Div")
+                .SetInput("A", "magnitude_squared")
+                .SetConstant("B", TensorFloat.CreateFromArray(new List<long>(), new float[] { dftSize }))
+                .SetOutput("C", "power_frames"))
+              .Operators.Add(new Operator("MelWeightMatrix", MS_EXPERIMENTAL_DOMAIN)
+                .SetConstant("num_mel_bins", TensorInt64Bit.CreateFromArray(new List<long>(), new long[] { nMelBins }))
+                .SetConstant("dft_length", TensorInt64Bit.CreateFromArray(new List<long>(), new long[] { dftSize }))
+                .SetConstant("sample_rate", TensorInt64Bit.CreateFromArray(new List<long>(), new long[] { samplingRate }))
+                .SetConstant("lower_edge_hertz", TensorFloat.CreateFromArray(new List<long>(), new float[] { 0 }))
+                .SetConstant("upper_edge_hertz", TensorFloat.CreateFromArray(new List<long>(), new float[] { (float)(samplingRate / 2.0) }))
+                .SetOutput("output", "mel_weight_matrix"))
+              .Operators.Add(new Operator("Reshape")
+                .SetInput("data", "power_frames")
+                .SetConstant("shape", TensorInt64Bit.CreateFromArray(new List<long>() { 2 }, new long[] { batchSize * nDfts, onesidedDftSize }))
+                .SetOutput("reshaped", "reshaped_output"))
+              .Operators.Add(new Operator("MatMul")
+                .SetInput("A", "reshaped_output")
+                .SetInput("B", "mel_weight_matrix")
+                .SetOutput("Y", "mel_spectrogram"))
+              .Operators.Add(new Operator("Reshape")
+                .SetInput("data", "mel_spectrogram")
+                .SetConstant("shape", TensorInt64Bit.CreateFromArray(new List<long>() { 4 }, melSpectrogramShape))
+                .SetOutput("reshaped", "Output.MelSpectrogram"));
+
+            var model = builder.CreateModel();
+
+            LearningModelSession session = new LearningModelSession(model);
+            LearningModelBinding binding = new LearningModelBinding(session);
+
+            binding.Bind("Input.TimeSignal", TensorFloat.CreateFromArray(signalShape, signal.ToArray()));
+
+            // Bind output
+            var outputImage = new VideoFrame(
+                BitmapPixelFormat.Bgra8,
+                nMelBins,
+                nDfts);
+            //var outputImage = new Bitmap(
+            //    nMelBins,
+            //    nDfts);
+
+
+            binding.Bind("Output.MelSpectrogram", outputImage);
+
+            // Evaluate
+            var sw = Stopwatch.StartNew();
+            var result = session.Evaluate(binding, "");
+            sw.Stop();
+            Console.WriteLine("Evaluate Took: %f\n", sw.ElapsedMilliseconds);
+
+            //return outputImage.SoftwareBitmap;
+            return outputImage.SoftwareBitmap;
+
+            // check the output video frame object by saving output image to disk
```

Comment:
Maybe move the save into a separate funciton?

Reply:
Moved in the WinUI project